apps/stripe/src/modules/saleor/saleor-payment-method-details.ts

```diff
@@ -0,0 +1,90 @@
+import { err, ok, Result } from "neverthrow";
+import Stripe from "stripe";
+
+import { Paymentmethoddetails } from "@/generated/app-webhooks-types/transaction-process-session";
+import { PaymentMethodDetailsInput } from "@/generated/graphql";
+import { BaseError } from "@/lib/errors";
+
+export type SaleorPaymentMethodDetailsErrorType = InstanceType<
+  typeof SaleorPaymentMethodDetails.SaleorPaymentMethodDetailsError
+>;
+
+export class SaleorPaymentMethodDetails {
+  private stripePaymentMethod: Stripe.PaymentMethod;
+
+  private constructor(stripePaymentMethod: Stripe.PaymentMethod) {
+    this.stripePaymentMethod = stripePaymentMethod;
+  }
+
+  static SaleorPaymentMethodDetailsError = BaseError.subclass("SaleorPaymentMethodDetailsError", {
+    props: {
+      _internalName: "SaleorPaymentMethodDetails.SaleorPaymentMethodDetailsError" as const,
+    },
+  });
+
+  static createFromStripe(
+    paymentMethod: string | Stripe.PaymentMethod | null,
+  ): Result<SaleorPaymentMethodDetails, SaleorPaymentMethodDetailsErrorType> {
+    if (!paymentMethod) {
+      return err(
+        new SaleorPaymentMethodDetails.SaleorPaymentMethodDetailsError("Payment method is null"),
+      );
+    }
+
+    if (typeof paymentMethod === "string") {
+      return err(
+        new SaleorPaymentMethodDetails.SaleorPaymentMethodDetailsError(
+          "Payment method is a string",
+        ),
+      );
+    }
```

Comment:
I don't get it. So why do you accept it as string in argument, but return error in such case?

Reply:
it can come from Stripe - they can send either: `string (not expanded) | object (expanded) & null`